packages/synapse-react-client/src/lib/utils/hooks/useLogin.ts

```diff
@@ -98,28 +101,80 @@ export default function useLogin(
     }
   }
 
-  const submitUsernameAndPassword: UseLoginReturn['submitUsernameAndPassword'] =
-    async (username, password) => {
-      setErrorMessage(undefined)
-      try {
-        const authenticationReceipt = localStorage.getItem(
-          AUTHENTICATION_RECEIPT_LOCALSTORAGE_KEY,
-        )
-        const data = await SynapseClient.login(
-          username,
-          password,
-          authenticationReceipt,
-        )
-        if ('errorCode' in data) {
+  const {
+    mutateAsync: mutateLoginWithUsernameAndPassword,
+    isLoading: isLoadingLoginWithUsernameAndPassword,
+  } = useMutation<
```

Comment:
The easiest way to track loading state in an asynchronous request is `useMutation`. I don't think we'd reuse these, and they don't use query keys, so it made more sense to me to keep them here, rather than putting them with the rest of the react-query hooks